flink-end-to-end-tests/flink-metrics-availability-test/src/test/java/org/pache/flink/metrics/tests/MetricsAvailabilityITCase.java

```diff
@@ -0,0 +1,160 @@
+import org.apache.flink.runtime.rest.messages.taskmanager.TaskManagersHeaders;
+import org.apache.flink.runtime.rest.messages.taskmanager.TaskManagersInfo;
+import org.apache.flink.tests.util.FlinkDistribution;
+import org.apache.flink.util.TestLogger;
+import org.apache.flink.util.function.SupplierWithException;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * End-to-end test for the availability of metrics.
+ */
+public class MetricsAvailabilityITCase extends TestLogger {
+
+	private static final String HOST = "localhost";
+	private static final int PORT = 8081;
+
+	@Rule
+	public final FlinkDistribution dist = new FlinkDistribution();
+
+	@Nullable
+	private static ScheduledExecutorService scheduledExecutorService = null;
+
+	@BeforeClass
+	public static void startExecutor() {
+		scheduledExecutorService = Executors.newScheduledThreadPool(4);
+	}
+
+	@AfterClass
+	public static void shutdownExecutor() {
+		if (scheduledExecutorService != null) {
+			scheduledExecutorService.shutdown();
+		}
+	}
+
+	@Test
+	public void testReporter() throws Exception {
+		dist.startFlinkCluster();
+
+		final RestClient restClient = new RestClient(RestClientConfiguration.fromConfiguration(new Configuration()), scheduledExecutorService);
+
+		checkJobManagerMetricAvailability(restClient);
+
+		final Collection<ResourceID> taskManagerIds = getTaskManagerIds(restClient);
+
+		for (final ResourceID taskManagerId : taskManagerIds) {
+			checkTaskManagerMetricAvailability(restClient, taskManagerId);
+		}
+	}
+
+	private static void checkJobManagerMetricAvailability(final RestClient restClient) throws Exception {
+		final JobManagerMetricsHeaders headers = JobManagerMetricsHeaders.getInstance();
+		final JobManagerMetricsMessageParameters parameters = headers.getUnresolvedMessageParameters();
+		parameters.metricsFilterParameter.resolve(Collections.singletonList("numRegisteredTaskManagers"));
+
+		fetchMetric(() ->
+				restClient.sendRequest(HOST, PORT, headers, parameters, EmptyRequestBody.getInstance()),
+			getMetricNamePredicate("numRegisteredTaskManagers"));
+	}
+
+	private static Collection<ResourceID> getTaskManagerIds(final RestClient restClient) throws Exception {
+		final TaskManagersHeaders headers = TaskManagersHeaders.getInstance();
+
+		final TaskManagersInfo response = fetchMetric(() ->
+				restClient.sendRequest("localhost",
```

Comment:
Move the `localhost` to new line (to have one argument per line) and replace it with `HOST`.